src/vm.js

```diff
@@ -766,25 +768,33 @@ VM.prototype.setMetadata = function(metadata, callback) {
       return;
     }
 
-    const newMetadata = {
+    const request = {
       fingerprint: currentMetadata.metadata.fingerprint,
       items: [],
     };
 
-    for (const prop in metadata) {
-      if (metadata.hasOwnProperty(prop)) {
-        newMetadata.items.push({
-          key: prop,
-          value: metadata[prop],
-        });
+    const metadataJSON = (currentMetadata.metadata.items || []).reduce((metadataJSON, keyValPair) => {
+      metadataJSON[keyValPair.key] = keyValPair.value;
+      return metadataJSON;
+    }, {});
+
+    const newMetadataJSON = extend(metadataJSON, metadata);
```

Comment:
I'm not actually sure what the minimum Node/JS version is for this library, but the lack of nested fields means we could use `Object.assign` instead.